htap.go

```diff
@@ -22,5 +25,33 @@ func (qcc *queryContextCache) init() *queryContextCache {
 func (qcc *queryContextCache) add(qces ...queryContextEntry) {
 	qcc.mutex.Lock()
 	defer qcc.mutex.Unlock()
-	qcc.entries = append(qcc.entries, qces...)
+	var newQcesBasedOnPriorities []queryContextEntry
+	for _, newQce := range qces {
+		logger.Debugf("adding query context: %v", newQce)
+		newQceProcessed := false
+		for existingQceIdx, existingQce := range qcc.entries {
+			if newQce.ID == existingQce.ID {
+				newQceProcessed = true
+				if newQce.Timestamp > existingQce.Timestamp {
+					qcc.entries[existingQceIdx] = newQce
+				} else if newQce.Timestamp == existingQce.Timestamp {
+					if newQce.Priority != existingQce.Priority {
+						qcc.entries[existingQceIdx] = newQce
+					}
+				}
+			} else {
+				if newQce.Priority == existingQce.Priority {
+					qcc.entries[existingQceIdx] = newQce
+					newQceProcessed = true
```

Comment:
Don't we need any break/exit after setting newQceProcessed to true?
Lets assume scenario: newQce has same ID as one qcc element and same Priority as other qcc element. If I'm not wrong it will replace two qcc elements.

Reply:
Confirmed with the server team (@sfc-gh-lqin ) that this is not possible scenario.